Bug in the preparation of the simulation of some stateful cells in some cases

### Version

Yosys 0.37+21 (git sha1 8649e3066, clang 10.0.0-4ubuntu1 -fPIC -Os)

### On which OS did this happen?

Linux

### Reproduction Steps

Hi there!
In some cases with stateful cells, the compilation step fails.
[I created a dedicated repository](https://github.com/flaviens/cxxrtl-b2) to help you reproduce this issue.
It does not seem related to #4136 , I am working with the last Yosys version.

Thank you!
Flavien

### Expected Behavior

Successful compilation

### Actual Behavior

```
cxxrtl_obj_dir/cxxrtl.cpp:118:90: error: ‘struct cxxrtl::slice_expr<cxxrtl::value<2>, 0, 0>’ has no member named ‘update’
  118 |  p_top__i_2e_celloutsig__0z.next.slice<0>() = p_top__i_2e_celloutsig__0z.next.slice<0>().update(value<1>{0x1u}, (p_top__i_2e_clkin__data.slice<96>().val() ? value<1>{0u} : (p_top__i_2e_clkin__data.slice<64>().val() ? value<1>{0x1u} : value<1>{0u})));
      |                                                                                          ^~~~~~
cxxrtl_obj_dir/cxxrtl.cpp:119:90: error: ‘struct cxxrtl::slice_expr<cxxrtl::value<2>, 0, 0>’ has no member named ‘update’
  119 |  p_top__i_2e_celloutsig__0z.next.slice<0>() = p_top__i_2e_celloutsig__0z.next.slice<0>().update(value<1>{0u}, (p_top__i_2e_clkin__data.slice<96>().val() ? value<1>{0x1u} : value<1>{0u}));
```

--- a/backends/cxxrtl/cxxrtl_backend.cc
+++ b/backends/cxxrtl/cxxrtl_backend.cc
@@ -1446,7 +1446,7 @@ struct CxxrtlWorker {
 				f << indent;
 				dump_sigspec_lhs(cell->getPort(ID::Q));
 				f << " = ";
-				dump_sigspec_lhs(cell->getPort(ID::Q));
+				dump_sigspec_rhs(cell->getPort(ID::Q));
 				f << ".update(";
 				dump_const(RTLIL::Const(RTLIL::S1, cell->getParam(ID::WIDTH).as_int()));
 				f << ", ";
@@ -1458,7 +1458,7 @@ struct CxxrtlWorker {
 				f << indent;
 				dump_sigspec_lhs(cell->getPort(ID::Q));
 				f << " = ";
-				dump_sigspec_lhs(cell->getPort(ID::Q));
+				dump_sigspec_rhs(cell->getPort(ID::Q));
 				f << ".update(";
 				dump_const(RTLIL::Const(RTLIL::S0, cell->getParam(ID::WIDTH).as_int()));
 				f << ", ";

